packages/nouns-contracts/test/descriptorV3.test.ts

```diff
@@ -0,0 +1,136 @@
+import { LongestPart } from './types';
+import { deployNounsDescriptorV3, populateDescriptorV2 } from './utils';
+import { ethers } from 'hardhat';
+
+chai.use(solidity);
+const { expect } = chai;
+
+const eyesred =
+  '0x000b17100703000624010006240300012402020265012401000124020202650524020202650324020202650224020001240202026501240100012402020265022402000124020202650124010001240202026501240300062401000624';
+
+const hiprose =
+  '0x000b1710070300062101000621030001210202022401210100012102020224052102020224032102020224052102020224032102020224022102000121020202240121010001210202022401210300062101000621';
+
+describe('NounsDescriptorV3', () => {
+  let nounsDescriptor: NounsDescriptorV3;
+  let snapshotId: number;
+
+  const glassesTraitIndex = 1;
+
+  const updateGlassesTraits = async () => {
+    const newEncodedCompressedTrait = ProposalImages[0].glasses?.encodedCompressed ?? '';
+    const decompressedLength = ProposalImages[0].glasses?.originalLength ?? 0;
+    const itemCount = ProposalImages[0].glasses?.itemCount ?? 0;
+
+    const glassesCountBefore = await nounsDescriptor.glassesCount();
+
+    console.log(`glassesCountBefore: ${glassesCountBefore}`);
+    await nounsDescriptor.updateGlasses(newEncodedCompressedTrait, decompressedLength, itemCount, {
+      gasLimit: 30000000,
+    });
+
+    const glassesCountAfter = await nounsDescriptor.glassesCount();
+    const glassesAtIdex = await nounsDescriptor.glasses(glassesTraitIndex);
+
+    console.log(`glassesCountAfter: ${glassesCountAfter}. Index0: ${glassesAtIdex}`);
+
+    expect(glassesAtIdex).to.equal(hiprose);
+  };
+
+
+  const checkGlassesAfterUpdate = async () => {
+    const traitIndex = 1;
+    const glassesTrait = await nounsDescriptor.glasses(traitIndex, {
+      gasLimit: 30000000,
+    });
+
+    const glassesCountAfter = await nounsDescriptor.glassesCount();
+    console.log(`glassesCountAfter: ${glassesCountAfter}. trait:${glassesTrait}`);
+
+    expect(glassesTrait).to.equal(hiprose);
+  };
+
+  const part: LongestPart = {
+    length: 0,
+    index: 0,
+  };
+  const longest: Record<string, LongestPart> = {
+    bodies: part,
+    accessories: part,
+    heads: part,
+    glasses: part,
+  };
+
+  before(async () => {
+    nounsDescriptor = await deployNounsDescriptorV3();
+
+    for (const [l, layer] of Object.entries(ImageData.images)) {
+      for (const [i, item] of layer.entries()) {
+        if (item.data.length > longest[l].length) {
+          longest[l] = {
+            length: item.data.length,
+            index: i,
+          };
+        }
+      }
+    }
+
+    await populateDescriptorV2(nounsDescriptor);
+    await updateGlassesTraits();
+  });
+
+  beforeEach(async () => {
+    snapshotId = await ethers.provider.send('evm_snapshot', []);
+  });
+
+  afterEach(async () => {
+    await ethers.provider.send('evm_revert', [snapshotId]);
+  });
+
+  it('updates all glasses with updated trait at index 0', async () => {
```

Comment:
@EricDevito the description of the test isn't very clear IMO. can you please consider updating it such that it clarifies what you are testing exactly. for example, are you replacing the art of the first glasses item from X to Y?

Reply:
And how do you know that the update didn't mess up other glasses?